### ISSUE TITLE
Improved folderdocumentation and Readme.md support

### ISSUE THREAD

**Issue by Nos-** [doxygen/doxygen#7587]
Hi, i am like to use README.md-files in several (sub-)folders of my projects. Since Github and others renders such markdown files below the list of files, everyone can get an quick overview of the content of the folder.

But there is some trouble with similar named markdown files in Doxygen till now. What do you think about folder based readme files which are automatically usable for folder selfdescription?

**Comment by albert-github** [comment]
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem?  Please don't add external links as they might not be persistent.
- Please also specify the doxygen version used.

**Comment by Nos-** [comment]
Yes of course, i am using Doxygen 1.8.15  on Win/ Linux. Find the example and config attached.
[Doxygen-example.zip](https://github.com/doxygen/doxygen/files/4209688/Doxygen-example.zip)

**Comment by albert-github** [comment]
Looks like that setting `USE_MDFILE_AS_MAINPAGE = ./README.md` instead if `USE_MDFILE_AS_MAINPAGE = README.md` solves a number of problems.

Also the reference in `src\FolderB\README.md` is not correct. This should point to the right `README.md` file `@mainpage` is not a reference, though here you might run into some other problems (see: "Relative markdown file reference", i.e. proposed pull request  #7032).

Also have a look at `GENERATE_TREEVIEW=YES`

Or did I misunderstand the problem?

**Comment by Nos-** [comment]
Thanks for the advice. I got some trouble in case there are several Readme.md files in several directories (maybe doxygen can not distinguish them). The possibility of crossreferencing would be nice but is not so important to me.
I tried the tree-view (which is really nice) but the files overview is already enough for me. Is there any possibility to use (Readme-)markdownfiles as folderdescription?

**Comment by albert-github** [comment]
Probably when you use the `\dir` command.
Question is a bit unclear to me though.
